server/rpc/pub_sub.go

```diff
@@ -17,16 +17,18 @@ type subscriberSet map[Subscriber]struct{}
 
 // PubSub - pub/sub interface for message senders
 type PubSub struct {
-	lock   sync.Mutex
-	subMap map[string]subscriberSet      // Subscription ID -> Subscribers
-	idMap  map[Subscriber]util.StringSet // Subscriber -> Subscription IDs
+	lock     sync.Mutex
```

Comment:
Please also change this to *sync.Mutex since mutexes aren't supposed to be copied. It works if there's only one instance, but if it's moved around it will get copied